src/boot/store.js

```diff
@@ -204,6 +227,44 @@ function listMiddleware() {
     // functions of `dispatch` and `state`.  See docs:
     //   https://github.com/reduxjs/redux-thunk
     thunkMiddleware,
+
+    /**
+     * Drop an action from an account that is not the active account.
+     *
+     * Will only be given a plain-object action as input, because of
+     * its placement after `thunkMiddleware`.
+     *
+     * If the action has the special `initiatingIdentity` field, check
+     * it against the identity of the currently active account, and
+     * drop the action if it's not a match.
+     *
+     * This can't go in the reducers because each reducer is only
+     * aware of the state it's in charge of, and we need to grab info
+     * on the active account, which is just in one part of the state.
+     *
+     * We could take care of this at all the dispatch sites, but it's
+     * easier to get this (privacy-relevant) check right in all those
+     * places if all we have to do is include the `initiatingIdentity`
+     * field.
+     */
+    ({ getState, dispatch }) => next => (action: Action) => {
+      // TODO: In `eventToAction`, stop spreading raw events straight
+      // from the server; we don't want any actions to include
+      // `initiatingIdentity` by mistake.
```

Comment:
FWIW I think we're actually pretty safe from this particular risk... because the server API systematically uses `snake_case`, not `camelCase` like this name. :slightly_smiling_face: 

(Though that spreading of raw events is certainly a thing we want to fix for other reasons!)